src/transform/FooterReadMore.js

```diff
@@ -0,0 +1,349 @@
+import './FooterReadMore.css'
+
+/**
+ * @typedef {function} SaveButtonClickHandler
+ * @param {!string} title
+ * @return {void}
+ */
+
+/**
+  * @typedef {function} TitlesShownHandler
+  * @param {!string[]} titles
+  * @return {void}
+  */
+
+/**
+   * Display fetched read more pages.
+   * @typedef {function} ShownReadMoreHandler
+   * @param {!Object[]} pages
+   * @param {!Document} document
+   * @param {!string} containerID
+   * @param {SaveButtonClickHandler} saveButtonClickHandler
+   * @param {TitlesShownHandler} titlesShownHandler
+   * @return {void}
+   */
+
+let _saveForLaterString = null
```

Comment:
Good news! I didn't need those globals after all. We were already calling a method from native land to set the save button icon state - I modified (& renamed) it to be passed the text to use for the current state of the save button at the same time.